src/main/java/io/hops/metadata/ndb/dalimpl/yarn/ResourceClusterJ.java

```diff
@@ -145,10 +145,6 @@ public void add(Resource resourceNode) throws StorageException {
   }
 
   private Resource createHopResource(ResourceDTO resourceDTO) {
-    if (resourceDTO == null) {
```

Comment:
With the current change, you will skip the null dtos, however, not sure if any other parts of the code you were dependent on returning null if not found. You need to check if you are dependent on returning null or not at the Context.

Reply:
createHopResource is called in findEntry and in createMap.
 - In findEntry it is called only if the dto is not null so this if will never be true.
 - In createMap it is followed by a hop.getId() which would throw a NullpointerException if we passed in this if.
So returning null is not supported by the current code.